bin/mixer_codegen.sh

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env bash
+
+WD=$(dirname $0)
+WD=$(cd $WD; pwd)
+ROOT=$(dirname $WD)
+
+set -e
+
+outdir=$ROOT
+file=$ROOT
+protoc="protoc-min-version -version=3.5.0"
+optimport=$ROOT
+
+while getopts 'f:o:p:i:' flag; do
+  case "${flag}" in
+    f) file+="/${OPTARG}" ;;
+    o) outdir="${OPTARG}" ;;
+    p) protoc="${OPTARG}" ;;
+    i) optimport+=/"${OPTARG}" ;;
+    *) error "Unexpected option ${flag}" ;;
+  esac
+done
+
+# echo "outdir: ${outdir}"
+
+# Ensure expected GOPATH setup
+if [ $ROOT != "${GOPATH-$HOME/go}/src/istio.io/istio" ]; then
+       echo "Istio not found in GOPATH/src/istio.io/"
+       exit 1
+fi
+
+if [ ! -e $GOPATH/bin/protoc-gen-gogoslick ]; then
+echo "Installing protoc-gen-gogoslick..."
+pushd $ROOT
+go install "./vendor/github.com/gogo/protobuf/protoc-gen-gogoslick"
+popd
+echo "Done."
+fi
+
+if [ ! -e $GOPATH/bin/protoc-min-version ]; then
+echo "Installing protoc-min-version..."
+pushd $ROOT
+go install "./vendor/github.com/gogo/protobuf/protoc-min-version"
+popd
+echo "Done."
+fi
+
+SHA=c8c975543a134177cc41b64cbbf10b88fe66aa1d
+GOOGLEAPIS_URL=https://raw.githubusercontent.com/googleapis/googleapis/${SHA}
+
+if [ ! -e ${ROOT}/vendor/github.com/googleapis/googleapis ]; then
+echo "Pull down source protos from googleapis..."
+
+mkdir -p ${ROOT}/vendor/github.com/googleapis/googleapis
+
+# all the google_rpc protos
+mkdir -p ${ROOT}/vendor/github.com/googleapis/googleapis/google/rpc
+curl -sS ${GOOGLEAPIS_URL}/google/rpc/status.proto > ${ROOT}/vendor/github.com/googleapis/googleapis/google/rpc/status.proto
```

Comment:
This is a reproducibility risk. @costinm @geeknoid @rshriram I think we need to commit protos alongside .pb.go files to https://github.com/istio/gogo-genproto/tree/master/googleapis/google/rpc

Reply:
Committing the protos SGTM. I already have to follow up with more pb.gos for that repo anyway.